src/generate.ts

```diff
@@ -30,7 +30,7 @@ export function setupStaticGeneration (nuxt: any, options: ModuleOptions) {
 
         staticImages[url] = options.staticFilename.replace(/\[(\w+)]/g, (match, key) => params[key] || match)
       }
-      return staticImages[url]
+      return joinURL(nuxt.options.app.basePath, staticImages[url])
```

Comment:
@danielroe Would you please double-check this? Since we are generating into `dist/_nuxt/image` shouldn't it be `cdnURL`?

Reply:
Yes, it should take that into account. Something like:
```suggestion
      return joinURL(nuxt.options.app.cdnURL || nuxt.options.app.basePath, staticImages[url])
```
Plus the following above (with something to replace the resulting `//`?):
```js
publicPath: nuxt.options.app.cdnURL ? '/' : withoutTrailingSlash(nuxt.options.build.publicPath)
```

For reference:
https://github.com/nuxt/nuxt.js/blob/ca020cf9cb4f88a85b06d666c846903d053a8e09/packages/config/src/options.js#L457-L461

Reply:
Updated with suggestion and tests still passing.